hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -588,6 +588,19 @@ protected void runTableServicesInline(HoodieTable table, HoodieCommitMetadata me
         metadata.addMetadata(HoodieClusteringConfig.SCHEDULE_INLINE_CLUSTERING.key(), "true");
         inlineScheduleClustering(extraMetadata);
       }
+
+      // if clustering is disabled, but we might need to rollback any inflight clustering when clustering was enabled previously.
+      if (!config.inlineClusteringEnabled() && !config.isAsyncClusteringEnabled() && !config.scheduleInlineClustering()
```

Comment:
@nsivabalan i do have vague remembrance that we touched upon this prior, so pardon me for repeating myself if so: 

I don't think we should be rolling back any pending clustering after disabling -- instead we should actually do the following

 - Disable just the clustering _planning_ phase when config is flipped
 - Predicate clustering execution solely on whether it has a plan -- it should ignore whether it's enabled or not and only check whether there's a plan for it

Reply:
this is already the case. The issue we are trying to solve here is, 
if the replace commit requested is left in the data timeline, then metadata table compaction is stopped. 
thats why.

Reply:
@nsivabalan why do we need to roll it back then?